src/api/AppActivity.js

```diff
@@ -0,0 +1,160 @@
+        if (!id) {
+            throw new Error('Missing file id!');
+        }
+
+        return `${this.getUrl()}/${id}`;
+    }
+
+    /**
+     * Generic success handler
+     *
+     * @param {AppActivityItems} data - the response data
+     */
+    successHandler = ({ entries = [] }: AppActivityItems): void => {
+        if (this.isDestroyed() || typeof this.successCallback !== 'function') {
+            return;
+        }
+
+        const activityEntries = entries.map(this.mapAppActivityItem);
+        this.successCallback({ entries: activityEntries, total_count: activityEntries.length });
+    };
+
+    /**
+     * Generic error handler
+     *
+     * @param {AxiosError} error - the error response
+     */
+    errorHandler = (error: $AxiosError<any>): void => {
+        if (this.isDestroyed() && typeof this.errorCallback !== 'function') {
+            return;
+        }
+
+        this.permissions = {};
+        const { response } = error;
+
+        // In the case of a 404, the enterprise does not have App Activities enabled.
+        // Show no App Activity
+        if (response.status === HTTP_STATUS_CODE_NOT_FOUND) {
+            this.successHandler({});
+        } else {
+            super.errorHandler(error);
+        }
+    };
+
+    /**
+     * API for fetching App Activity on a file
+     *
+     * @param {string} id - the file id
+     * @param {BoxItemPermission} permissions - Permissions to attach to the app activity items. Determines if it can be deleted.
+     * @param {Function} successCallback - the success callback
+     * @param {Function} errorCallback - the error callback
+     * @param {number} [limit] - the max number of app activity items to return.
+     *
+     * @returns {void}
+     */
+    getAppActivity(
+        id: string,
+        permissions: BoxItemPermission,
+        successCallback: Function,
+        errorCallback: ElementsErrorCallback,
+        limit?: number,
+    ): void {
+        const requestData = {
+            item_id: id,
+            item_type: 'file',
+            fields: APP_ACTIVITY_FIELDS_TO_FETCH.toString(),
+        };
+
+        this.permissions = permissions;
+
+        this.markerGet({
+            id,
+            limit,
+            successCallback,
+            errorCallback,
+            requestData,
+        });
+    }
+
+    /**
+     * Delete an app activity item
+     *
+     * @param {BoxItem} file - The Box file that App Activity is on
+     * @param {string} appActivityId - An AppActivity item id
+     * @param {Function} successCallback - The success callback
+     * @param {Function} errorCallback - The error callback
+     */
+    deleteAppActivity({
+        file,
+        appActivityId,
+        successCallback,
+        errorCallback,
+    }: {
+        appActivityId: string,
+        errorCallback: Function,
+        file: BoxItem,
+        successCallback: Function,
+    }): void {
+        this.errorCode = ERROR_CODE_DELETE_APP_ACTIVITY;
```

Comment:
I've seen in other areas we pass permissions into the delete* call and then invoke `checkApiCallValidity`, is that a pattern we should use here as well instead of storing the permissions in the API class?

Reply:
The reason we leave it out is because it is possible for the user to be missing permissions, but be able to delete the activity. This is true in the case that the user isn't an editor/owner of the file, but they're the one who made the app activity.